Logger.md

```diff
@@ -0,0 +1,107 @@
+# Logger
+
+`Logger` is an interface that helps decouple client classes from concrete implementations of a logging system.
+
+## Declaration
+
+```swift
+public enum LogLevel {
+    case info
+    case warning
+    case error
+}
+
+public protocol Logger {
+  func log(_ level: LogLevel, tag: String, message: String)
+  func print(tag: String, message: String)
+  func warning(tag: String, message: String) 
+  func error(tag: String, message: String)
+}
+```
+
+```kotlin
+// Kotlin
+TODO
+```
+
+## Usage
+
+```swift
+// Swift
+
+#if DEBUG
+  let logger = DeviceConsoleLogger() // This class is an example
+#else
+  let logger = BugfenderLogger() // This class is an example 
+#endif
+
+let manager = NetworkManager(logger) // This class is an example 
+
+```
+
+## Implementation
+
+```swift
+// Swift
+// Swift
+/// Information that gets logged can be labeled as one of the following informative labels.
+public enum LogLevel {
+    case info
+    case warning
+    case error
+}
+
+/// Abstracts concrete implementations of a logger system.
+public protocol Logger {
+    
+    /// Logs a String object using a given level
+    ///
+    /// - Parameters:
+    ///   - level: Type of log.
+    ///   - tag: An additional label to help categorise logs.
+    ///   - message: The message to be logged.
+    func log(_ level: LogLevel, tag: String, message: String)
```

Comment:
Would it be good to make the level with decorator default to `.info`, the tag an optional string and with a custom parameter value of nil, and the message without parameter decorator: 

`func log(level: LogLevel = .info, tag: String? = nil, _ message: String)`.

This way, if the user doesn't want to use a tag or not specify a level, can do it straight forward.

`MyLogger.log("Hello World")` 
`MyLogger.log(level: .warning, "Hello World")`
`MyLogger.log(level: .warning, tag: "Network", "Hello World")`

Same should apply to the 3 helper methods:

```
func print(tag: String? = nil, _ message: String)
func warning(tag: String? = nil, _ message: String)
func error(tag: String? = nil, _ message: String)

MyLogger.print("HelloWorld")
MyLogger.print(tag: "Network", "HelloWorld")

MyLogger.warning("HelloWorld")
MyLogger.warning(tag: "Network", "HelloWorld")

MyLogger.error("HelloWorld")
MyLogger.error(tag: "Network", "HelloWorld")
```

Reply:
remember to:
-  make `_ message: String`
- Apply the same changes to the main `log` method

Reply:
Honestly, adding a default tag value to the `log()` function when you have the `print()` that it's doing the same it's a bit nonsense but nevermind up to you.